swan-lake/development-tutorials/observability/supported-observability-tools-and-platforms/prometheus.md

```diff
@@ -0,0 +1,164 @@
+
+## Step 4 - Run the Ballerina service
+
+When Ballerina observability is enabled, the Ballerina runtime exposes internal metrics via an HTTP endpoint (`/metrics`) for metrics monitoring and the metrics will be published to Prometheus. Prometheus should be configured to scrape metrics from the metrics HTTP endpoint in Ballerina.
+
+Run the following command to start the Ballerina service.
+
+```
+$ bal run
+
+Compiling source
+
+Running executable
+
+ballerina: started Prometheus HTTP listener 0.0.0.0:9797
+```
+
+## Step 5 - Send requests
+ 
+Send requests to <http://localhost:8090/shop/products>.
+
+Example cURL commands:
+
+```
+$ curl -X GET http://localhost:8090/shop/products
+```
+```
+$ curl -X POST http://localhost:8090/shop/product \
+-H "Content-Type: application/json" \
+-d '{
+    "id": 4, 
+    "name": "Laptop Charger", 
+    "price": 50.00
+}'
+```
+```
+$ curl -X POST http://localhost:8090/shop/order \
+-H "Content-Type: application/json" \
+-d '{
+    "productId": 1, 
+    "quantity": 1
+}'
+```
+```
+$ curl -X GET http://localhost:8090/shop/order/0
+```
+
+## Step 6 - View metrics on the Prometheus server
+
+Go to <http://localhost:19090/> and check whether you can see the Prometheus graph.
+Ballerina metrics should appear in the Prometheus graph's metrics list when the Ballerina service is started.
+
+![Ballerina metrics listed in Prometheus](/learn/images/ballerina-metrics-listed-in-prometheus.png "Ballerina metrics listed in Prometheus")
+
+![Ballerina metric in graph format](/learn/images/ballerina-metrics-in-graph.png "Ballerina metric in graph format")
+
+You can also use the following command to get the metrics.
+
+```
+$ curl http://localhost:9797/metrics
+```
+
+## Set up Grafana
+
+[Grafana](https://grafana.com/) can be used to visualize Ballerina metrics provided for Prometheus. First, users need to set up the Ballerina project to observe metrics in Prometheus. Please follow the steps mentioned above.
+
+Let’s use Grafana to visualize metrics in a dashboard. For this, we need to install Grafana and configure Prometheus as a data source. Follow the steps below to configure Grafana.
+
+1. Start Grafana as a Docker container with the command below.
+
+    ```
+    $ docker run -d --name=grafana -p 3000:3000 grafana/grafana
+    ```
+    For more information, go to <a href="https://hub.docker.com/r/grafana/grafana/" target="_blank">Grafana in Docker Hub</a>.
+
+2. Go to <http://localhost:3000/> to access the Grafana dashboard running on Docker.
+
+3. Login to the dashboard with the default user, username: `admin` and password: `admin`
+
+4. Add Prometheus as a data source with the `Browser` access configuration as provided below.
+
+    ![Grafana Prometheus data source](/learn/images/grafana-prometheus-datasource.png "Grafana Prometheus data source")
```

Comment:
Since the steps we followed so far was to use docker image to run Prometheus, the Prometheus server url should be `http://host.docker.internal:19090/` Need to update the image.